fast-list.js

```diff
@@ -764,6 +813,82 @@ function reveal(list) {
   });
 }
 
+/**
+ * Detects if scrolling appears 'blocked'.
+ *
+ * We're BUSY if:
+ *
+ * The scroll position moved more than
+ * twice the height of the viewport
+ * since the last 'scroll' event.
+ *
+ * We're NO LONGER BUSY if:
+ *
+ * The scroll position moved less than
+ * *half* the viewport since the last
+ * 'scroll' event.
+ *
+ * If neither of these conditions are
+ * met we return the previous value.
+ *
+ * @param  {Boolean}  wasBusy
+ * @param  {Number}  moved  distance since last 'scroll'
+ * @param  {Number}  viewportHeight
+ * @return {Boolean}
+ */
```

Comment:
These were making my brain hurt a bit so gave theme a thorough explanation. Hope you don't mind :)